### PR TITLE
31 integrate ttop into ddc

### PR DESCRIPTION
**Pull request by rsvihladremio**

### CODE REVIEW

File: cmd/local/conf/defaults_test.go [hunk 1]
```diff
@@ -69,11 +69,14 @@ func TestSetViperDefaults(t *testing.T) {
...
 		{conf.KeyCollectSystemTablesExport, true},
 		{conf.KeyCollectWLM, true},
+		{conf.KeyCollectTtop, true},
 		{conf.KeyCollectKVStoreReport, true},
 		{conf.KeyDremioJStackTimeSeconds, defaultCaptureSeconds},
 		{conf.KeyDremioJFRTimeSeconds, defaultCaptureSeconds},
 		{conf.KeyNodeMetricsCollectDurationSeconds, defaultCaptureSeconds},
 		{conf.KeyDremioJStackFreqSeconds, 1},
+		{conf.KeyDremioTtopFreqSeconds, 1},
+		{conf.KeyDremioTtopTimeSeconds, 30},
```
Comment (markcurtis1970) [inline cmd/local/conf/defaults_test.go:79]: should we keep the same default as the thread dumps (60) ?

File: cmd/local/ttopcollect/ttopcollect.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	if err != nil {
+		return err
+	}
+
+	sjk := filepath.Join(t.tmpDir, "sjk.jar")
+	if err := os.WriteFile(sjk, data, 0600); err != nil {
+		return err
+	}
+
+	t.cmd = exec.Command("java", "-jar", sjk, "ttop", "-ri", fmt.Sprintf("%vs", interval), "-n", "100", "-p", fmt.Sprintf("%v", pid))
```
Comment (markcurtis1970) [inline cmd/local/ttopcollect/ttopcollect.go:71]: 100 threads? I personally think its a bit much but maybe you have a reason?

File: cmd/local/conf/defaults_test.go (hunk 1 above)
Comment (rsvihladremio) [reply cmd/local/conf/defaults_test.go:79]: it is actually just shoud use the variable will fix that

File: cmd/local/ttopcollect/ttopcollect.go (hunk 2 above)
Comment (rsvihladremio) [reply cmd/local/ttopcollect/ttopcollect.go:71]: it is but it approaches getting all threads on bigger servers with 64 cores. but we could lower it

Comment (markcurtis1970) [reply cmd/local/ttopcollect/ttopcollect.go:71]: ok thats fine good point lets keep at 100

**Review by markcurtis1970 — APPROVED** [review]

**Pull request merged**